### PR TITLE
Force fetching all pairs

### PR DESCRIPTION
**Pull request by luzzif**
Always fetch all pairs on each network to be shown in the pairs page. The current implementation limits the fetched pairs to 100. This is already an issue since today we have 119 pairs on xDai. 19 of those pairs are not shown as of now, including WBTCWETH, with over 500k USD in liquidity. This PR aims to fix this by implementing a paginated fetch, where each page contains up to 1k elements (future-proof).

### CODE REVIEW

File: src/hooks/useAllPairsWithNonExpiredLiquidityMiningCampaignsAndLiquidityAndStakingIndicator.ts [hunk 1]
```diff
@@ -1,18 +1,21 @@
...
 import { useActiveWeb3React } from '.'
 import { SubgraphLiquidityMiningCampaign } from '../apollo'
 import { useAllTokensFromActiveListsOnCurrentChain } from '../state/lists/hooks'
 import { toLiquidityMiningCampaigns } from '../utils/liquidityMining'
 import { useNativeCurrency } from './useNativeCurrency'
+import { immediateSubgraphClients } from '../apollo/client'
+
+const PAGE_SIZE = 1000
 
 const QUERY = gql`
```
Comment (adamazad) [inline src/hooks/useAllPairsWithNonExpiredLiquidityMiningCampaignsAndLiquidityAndStakingIndicator.ts:16]: Just a thought for a future version. We could put all queries into one place. In Aqua, they're all in `src/subgraph/queries.ts`.

File: src/hooks/useAllPairsWithNonExpiredLiquidityMiningCampaignsAndLiquidityAndStakingIndicator.ts [hunk 2]
```diff
@@ -104,24 +107,53 @@ export function useAllPairsWithNonExpiredLiquidityMiningCampaignsAndLiquidityAnd
...
+          const result = await immediateSubgraphClients[chainId].request<QueryResult>(QUERY, {
+            lowerTimeLimit: memoizedLowerTimeLimit,
+            userId: subgraphAccountId,
+            lastId,
+            pageSize: PAGE_SIZE
+          })
+          pairs.push(...result.pairs)
+          lastId = result.pairs[result.pairs.length - 1].address
+          if (result.pairs.length < PAGE_SIZE) break
+        }
```
Comment (adamazad) [inline src/hooks/useAllPairsWithNonExpiredLiquidityMiningCampaignsAndLiquidityAndStakingIndicator.ts:135]: Smart!

**Review by adamazad — APPROVED** [review]
Great job!

File: src/hooks/useAllPairsWithNonExpiredLiquidityMiningCampaignsAndLiquidityAndStakingIndicator.ts [hunk 3]
```diff
@@ -104,24 +107,53 @@ export function useAllPairsWithNonExpiredLiquidityMiningCampaignsAndLiquidityAnd
...
+  useEffect(() => {
+    let cancelled = false
+    const fetchData = async () => {
+      if (!chainId) return
+      const pairs = []
+      let lastId = ''
+      setLoadingPairs(true)
+      setPairs([])
+      try {
+        while (1) {
```
Comment (Mi-Lan) [inline src/hooks/useAllPairsWithNonExpiredLiquidityMiningCampaignsAndLiquidityAndStakingIndicator.ts:125]: Wondering whats the reason for this while loop?

**Review by Mi-Lan — APPROVED** [review]
Left one comment other that that LGTM!

**Review by pimato — APPROVED** [review]
works great, I see all pools on xDai now.

File: src/hooks/useAllPairsWithNonExpiredLiquidityMiningCampaignsAndLiquidityAndStakingIndicator.ts (hunk 3 above)
Comment (luzzif) [reply src/hooks/useAllPairsWithNonExpiredLiquidityMiningCampaignsAndLiquidityAndStakingIndicator.ts:125]: It's basically to make sure to fetch ALL the pairs. It breaks as soon as we encounter a non-full page. Until then, new paginated requests are made to fetch everything.

**Pull request merged**